internal/http/services/cback/cback.go

```diff
@@ -0,0 +1,285 @@
+		return
+	}
+
+	snapshotList, err := s.listSnapshots(user.Username, resp.ID)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if resp.Substring != "" {
+		ssID, searchPath = s.pathTrimmer(snapshotList, resp)
+
+		if ssID == "" {
+			http.Error(w, errtypes.NotFound("cback: snapshot not found").Error(), http.StatusNotFound)
+			return
+		}
+
+		err = s.checkFileType(resp.ID, ssID, user.Username, searchPath, resp.Source)
+
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusNotFound)
+			return
+		}
+
+		structbody := &requestTemp{
+			BackupID: resp.ID,
+			Snapshot: ssID,
+			Pattern:  searchPath,
+		}
+
+		jbody, err := json.Marshal(structbody)
+
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		resp, err := s.getRequest(user.Username, url, http.MethodPost, bytes.NewBuffer(jbody))
+
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		defer resp.Close()
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+		err = json.NewEncoder(w).Encode(resp)
+
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		if _, err := io.Copy(w, resp); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+	} else {
+
+		err = errtypes.NotFound("cback: resource not found")
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+}
+
+func (s *svc) handleListJobs(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	user, inContext := ctxpkg.ContextGetUser(ctx)
+
+	if !inContext {
+		http.Error(w, errtypes.UserRequired("no user found in context").Error(), http.StatusInternalServerError)
+		return
+	}
+
+	url := s.conf.APIURL + "/restores/"
+
+	resp, err := s.getRequest(user.Username, url, http.MethodGet, nil)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	defer resp.Close()
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusCreated)
+	err = json.NewEncoder(w).Encode(resp)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if _, err := io.Copy(w, resp); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+}
+
+func (s *svc) handleRestoreStatus(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	user, inContext := ctxpkg.ContextGetUser(ctx)
+
+	if !inContext {
+		http.Error(w, errtypes.UserRequired("no user found in context").Error(), http.StatusInternalServerError)
+		return
+	}
+
+	restoreID := chi.URLParam(r, "restore_id")
+
+	url := s.conf.APIURL + "/restores/" + restoreID
+	resp, err := s.getRequest(user.Username, url, http.MethodGet, nil)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
Here if you request for a restore ID you don't have access to (because you don't have access to the associated backup job), from the APIs you will get 403 FORBIDDEN, and this is the error you should return back to the reva client.
Also, if the restore job does not exist, you will get 404 NOT FOUND.
All the other cases, you can return an Internal error